### PR TITLE
added new Cray* toolchain versions with dependency pinning

### PR DESCRIPTION
**Pull request by gppezzi**
This PR adds eb files for the new CrayGNU, CrayIntel and CrayCCE toolchains (with dep. pinning). For validation, it adds also easyconfig files for HPL.

Requires: EB 2.5.0 and [hpcugent/easybuild-easyblocks/pull/766](https://github.com/hpcugent/easybuild-easyblocks/pull/766)

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.7
See https://gist.github.com/7fb936ce407b0d1bd61b for a full test report.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/0a3ee80209310ac02651 for a full test report.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/41c34e4d4ef3bfd19354 for a full test report.

**Comment by gppezzi** [comment]
@boegel please let me know how to proceed, if you prefer that I make separate PRs for each toolchain or not (the .eb are very similar, there are only a few differences on the dependencies)

**Comment by pforai** [comment]
@gppezzi Individual TC PRs is fine, each with a matching HPL config that builds with that would be ideal.

**Comment by boegel** [comment]
@gppezzi: I wouldn't bother splitting this up, it's not really going to help much (if anything, it'll add more overhead, for the both of us)

The general rule-of-thumb is to keep easyconfig PRs below 10 new easyconfigs.

This is kind of an exception, since the easyconfigs are so similar to each other (in two groups).
The HPL easyconfigs are pretty much a no-brainer since they're (near) identical to each other but also to existing HPL easyconfigs (only exception is the extra patch in the CCE ones).

The others are simple enough to review in a single PR, since it's basically just a matter of the toolchain definition.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/27a760d46e7dfb24ed23 for a full test report.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.7
See https://gist.github.com/0beca8c3b1462ee1645f for a full test report.

**Comment by gppezzi** [comment]
@pforai It would be great if you could test these easyconfigs (using easyblock from [#766](https://github.com/hpcugent/easybuild-easyblocks/pull/766)) on any other Cray site...

Of course it is only expected to work out of the box if they have the PEs from 2015.06 and 2015.11 installed.

File: easybuild/easyconfigs/c/CrayIntel/CrayIntel-2015.06-XC.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+version = '2015.06-XC'
+
+homepage = 'http://docs.cray.com/books/S-9407-1511'
+description = """Toolchain using Cray compiler wrapper, using PrgEnv-gnu module (PE release: November 2015).\n"""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+dependencies = [
+    ('PrgEnv-intel/5.2.82', EXTERNAL_MODULE),
+    ('craype/2.4.0', EXTERNAL_MODULE),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CrayIntel/CrayIntel-2015.06-XC.eb]: you're missing a version-fixed `intel` here

File: easybuild/easyconfigs/c/CrayIntel/CrayIntel-2015.11-XC.eb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name = 'CrayIntel'
+version = '2015.11-XC'
+
+homepage = 'http://docs.cray.com/books/S-9407-1511'
+description = """Toolchain using Cray compiler wrapper, using PrgEnv-gnu module (PE release: November 2015).\n"""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+dependencies = [
+    ('PrgEnv-intel/5.2.82', EXTERNAL_MODULE),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CrayIntel/CrayIntel-2015.11-XC.eb]: here too, missing version-fixed `intel`

**Comment by boegel** [comment]
@gppezzi: does it makes sense to use the same `intel` version in both `2015.06` and `2015.11`?

Also, can you submit a new test report? Just for `CrayIntel` is fine, but I guess a full one is easier. ;)

**Comment by boegel** [comment]
@pforai: ping on a test report for this on non-CSCS Cray systems?

**Comment by gppezzi** [comment]
@boegel the intel version didn't seem to change from June to November. the release notes don't mention the minor version they provide and this is the only one from the 15 series we have. (Release notes from may still suggested to use 14.x but not in June)



Test report is on the way

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.7
See https://gist.github.com/ee653dbbe01c9dd85df7 for a full test report.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/4737cfb98962bba09408 for a full test report.

**Comment by boegel** [comment]
Despite some concerns w.r.t. the `intel` version used in the `CrayIntel` toolchain, this is good to go.

Thanks a lot for working on this @gppezzi, this is a huge step towards stable Cray support (cfr. https://github.com/hpcugent/easybuild-framework/issues/1390)!

**Comment by pforai** [comment]
Test report by @pforai
**FAILED**
Build succeeded for 4 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/bdce401c48fa3d4ee2d6 for a full test report.

**Comment by pforai** [comment]
Test report by @pforai
**FAILED**
Build succeeded for 9 out of 12 (12 easyconfigs in this PR)
Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/650e5060c847e6e50142 for a full test report.

**Comment by boegel** [comment]
@pforai: what's going on with the failed reports? Is that the `swap` issue for all toolchain deps?

**Pull request merged**